qiskit/pulse/builder.py

```diff
@@ -1197,6 +1197,7 @@ def _qubits_to_channels(*channels_or_qubits: Union[int, chans.Channel]) -> Set[c
     return channels
 
 
+@deprecate_func(since="1.0.0")
```

Comment:
```suggestion
@deprecate_func(since="0.46.0")
```
@mtreinish Is there any guide for deprecation? Do we need to open two PRs? One to https://github.com/Qiskit/qiskit/tree/stable/0.46 with deprecation warning and another one to main branch with code removal?

Reply:
Yes, you will need 2 PRs, one with the deprecation (to 0.46) and one with the removal (to 1.0 == `main`), and the deprecation warnings should say "since 0.46" (as suggested by @nkanazawa1989). In case you are considering it, I think that there's no need to close and re-open it against `stable/0.46`, we can merge it to main and backport it to 0.46. (now that I think about it, this might actually be the recommended course of action, because else we would have diverging histories, @mtreinish?). It might actually be a good idea to write this down and have a sort of "guide" we can refer people to, because I expect it to be confusing for many contributors.

Reply:
The pattern we've been following is just opening a PR on `stable/0.46` for the deprecation paired with a removal PR on `main` for 1.0.0. This was mostly done to minimize CI time, it's only 2 CI runs instead of 3 to deprecate and remove. They can also be pushed in parallel without any dependency between PRs.

But, I wouldn't say we're committed to that as a general requirement, doing both a deprecation to be backported to `stable/0.46` followed by an immediate removal PR to main would also work fine. The only complication is for backport you'd need to use a comment to manually tell mergify to backport to `stable/0.46`. For example, leaving a comment like: `@Mergifyio backport stable/0.46` after the PR merges. Using the stable backport potential tag in this case will trigger a backport to stable/0.45 which is incorrect for new deprecations.

In the interest of documenting this I pushed up https://github.com/Qiskit/qiskit/pull/11205 to document the branching versioning strategy and all the general deprecation procedures around this. But I didn't put in specifics around how to handle PRs for this situation. I think we can updating `CONTRIBUTING.md` after #11205 merges if we feel more guidance is needed on the specifics around this. In general backporting and stable branch policy will be getting more complicated in general moving forward as we support > 1 stable branch at a time (this is just the first example of it).